modules/light-clients/08-wasm/types/vm.go

```diff
@@ -10,20 +10,36 @@ import (
 
 	errorsmod "cosmossdk.io/errors"
 
+	"github.com/cosmos/cosmos-sdk/codec"
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 var (
-	WasmVM *wasmvm.VM
-	// Store key for 08-wasm module, required as a global so that the KV store can be retrieved
-	// in the ClientState Initialize function which doesn't have access to the keeper.
-	// The storeKey is used to check the code hash of the contract and determine if the light client
-	// is allowed to be instantiated.
-	WasmStoreKey  storetypes.StoreKey
+	WasmVM        *wasmvm.VM
 	VMGasRegister = NewDefaultWasmGasRegister()
+	// storeKeyMap stores the storeKey for the 08-wasm module. Using a single global storetypes.StoreKey fails in the context
+	// of tests with multiple test chains utilized. As such, we utilize a workaround involving a mapping from the chains codec
+	// to the storeKey which can be used to store a key per test chain.
+	// This is required as a global so that the KV store can be retrieved in the ClientState Initialize function which doesn't
+	// have access to the keeper. The storeKey is used to check the code hash of the contract and determine if the light client
+	// is allowed to be instantiated.
+	storeKeyMap = make(map[codec.BinaryCodec]storetypes.StoreKey)
```

Comment:
can we open an issue to remove this :sweat_smile:

Reply:
It is forever ingrained as a stain in my brain. Will do though, then I'll be ecstatic when the issue is closed.

Reply:
I think we can remove it now? Since the tm tests have been commented out?

Reply:
dropped like a sack of potatoes.